src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -97,40 +97,12 @@ namespace winrt::TerminalApp::implementation
         return S_OK;
     }
 
-    // Function Description:
-    // - Recursively check our commands to see if there's a keybinding for
-    //   exactly their action. If there is, label that command with the text
-    //   corresponding to that key chord.
-    // - Will recurse into nested commands as well.
-    // Arguments:
-    // - settings: The settings who's keybindings we should use to look up the key chords from
-    // - commands: The list of commands to label.
-    static void _recursiveUpdateCommandKeybindingLabels(CascadiaSettings settings,
-                                                        IMapView<winrt::hstring, Command> commands)
-    {
-        for (const auto& nameAndCmd : commands)
-        {
-            const auto& command = nameAndCmd.Value();
-            if (command.HasNestedCommands())
-            {
-                _recursiveUpdateCommandKeybindingLabels(settings, command.NestedCommands());
-            }
-            else
-            {
-                // If there's a keybinding that's bound to exactly this command,
-                // then get the keychord and display it as a
-                // part of the command in the UI.
-                // We specifically need to do this for nested commands.
-                const auto keyChord{ settings.ActionMap().GetKeyBindingForAction(command.ActionAndArgs().Action(), command.ActionAndArgs().Args()) };
-                command.RegisterKey(keyChord);
-            }
-        }
-    }
-
-    void TerminalPage::SetSettings(CascadiaSettings settings, bool needRefreshUI)
+    winrt::fire_and_forget TerminalPage::SetSettings(CascadiaSettings settings, bool needRefreshUI)
```

Comment:
The first two lines are:
```cpp
_settings = settings;
co_await wil::resume_foreground(Dispatcher());
```

The latter line indicates that the function is called from background threads? In that case the prior assignment in itself is a race condition already. You won't notice this in general when testing this on x86 as it offers strong(er) atomicity guarantees for anything that's <= pointer sized - like a WinRT struct for instance. If you were to run and test this on ARM64 you'd run into trouble much more quickly. You don't need to worry about assigning the value after the `co_await` as the capture-issue with coroutines only exists for lambdas. Function arguments and local variables are properly retained across `co_await`s.

Reply:
Oh and the code doesn't hold a weak-ref to `this`, so it might refer to deallocated data after the `co_await`. C++ is truly a terr...ific language. 💖 lol